### PR TITLE
Attempt at stream extensions and recording toggle

### PR DESCRIPTION
**Pull request by cs-endo**
Allows a -r flag to enable recording (off by default) and an argument to change where the stream is directed to. Instructions slightly unclear so this is not final

### CODE REVIEW

File: stream-nvenc.sh [hunk 1]
```diff
@@ -1,8 +1,33 @@
 #!/bin/bash
+
+# DEFAULTS
+rtmpServer="" # Set to default location
+
```
Comment (rmil) [inline stream-nvenc.sh:8]: I'd say split this variable into two; the base URL and then the default stream path?

File: stream-nvenc.sh [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+        -h|--help)
+                echo "[./stream-nvenc.sh -r] to enable recording"
+		echo "[./stream-nvenc.sh {stream}] to stream to $rtmpServer/{stream}]"
+                exit 0
+                ;;
+        -r)
+                toFile="-vcodec pores -acodec copy "'"'"Recordings/${filename}.mov"'"'""
```
Comment (rmil) [inline stream-nvenc.sh]: vcodec and acodec are the old ffmpeg ways to change the codec. If you swap it to: `-c:v prores -c:a copy` we'll be in business

File: stream-nvenc.sh [hunk 3]
```diff
@@ -1,8 +1,33 @@
...
+                ;;
+        *)
+                rtmpServer+="/$1"
+                break
+                ;;
+  esac
+done
+
+echo " Server: $rtmpServer"
+echo " Starting event: $filename"
```
Comment (rmil) [inline stream-nvenc.sh]: Perhaps, also explicitly indicate that it is recording?

**Review by rmil — CHANGES_REQUESTED** [review]
Seems pretty good! Just need to have a nosey at my comments and I'll try and get a test to see it works

File: stream-nvenc.sh (hunk 1 above)
Comment (cs-endo) [reply stream-nvenc.sh:8]: Can you give an example of what each could be?

Comment (rmil) [reply stream-nvenc.sh:8]: Something like:
```
rtmpServer="rtmp://stream.ystv.co.uk/"
defaultStream="internal/ob2"
```

**Pull request merged**